coderd/runtimeconfig/config_test.go

```diff
@@ -0,0 +1,123 @@
+package runtimeconfig_test
+
+import (
+	"testing"
+
+	"github.com/coder/serpent"
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/v2/coderd/coderdtest"
+	"github.com/coder/coder/v2/coderd/runtimeconfig"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/enterprise/coderd/coderdenttest"
+	"github.com/coder/coder/v2/enterprise/coderd/license"
+	"github.com/coder/coder/v2/testutil"
+)
+
+// TestConfig demonstrates creating org-level overrides for deployment-level settings.
+func TestConfig(t *testing.T) {
+	t.Parallel()
+
+	vals := coderdtest.DeploymentValues(t)
+	vals.Experiments = []string{string(codersdk.ExperimentMultiOrganization)}
+	adminClient, _, _, _ := coderdenttest.NewWithAPI(t, &coderdenttest.Options{
+		Options: &coderdtest.Options{DeploymentValues: vals},
+		LicenseOptions: &coderdenttest.LicenseOptions{
+			Features: license.Features{
+				codersdk.FeatureMultipleOrganizations: 1,
+			},
+		},
+	})
+	altOrg := coderdenttest.CreateOrganization(t, adminClient, coderdenttest.CreateOrganizationOptions{})
+
+	t.Run("panics unless initialized", func(t *testing.T) {
+		t.Parallel()
+
+		field := runtimeconfig.Entry[*serpent.String]{}
+		require.Panics(t, func() {
+			field.StartupValue().String()
+		})
+
+		field.Init("my-field")
+		require.NotPanics(t, func() {
+			field.StartupValue().String()
+		})
+	})
+
+	t.Run("simple", func(t *testing.T) {
+		t.Parallel()
+
+		ctx := testutil.Context(t, testutil.WaitShort)
+		store := runtimeconfig.NewInMemoryStore()
+		resolver := runtimeconfig.NewOrgResolver(altOrg.ID, runtimeconfig.NewStoreResolver(store))
+		mutator := runtimeconfig.NewOrgMutator(altOrg.ID, runtimeconfig.NewStoreMutator(store))
```

Comment:
Just another option, I haven't put much thought, so feel free to ignore.

Another syntax could be `runtimeconfig.NewStoreMutator(store).WithOrg(orgID)`.

Just using a chain syntax rather than a new `NewOrgMutator` function :man_shrugging:. I just don't imagine we will have use anything more than the StoreMutator.